### PR TITLE
Update samples so they have the correct instructions for both Windows…

### LINKED ISSUE #3522: Sample documentation doesn't work on Windows
Only two samples have correct README files with Windows instructions:
* helloworld
* host_verify

The remaining samples have README.md instructions that are Linux specific and don't even say so:
* local_attestation
* attested_tls
* data-sealing
* file-encryptor
* pluggable_allocator
* switchless
* remote_attestation

The samples do work using the instructions [here](https://github.com/shants/openenclave/blob/master/samples/README_Windows.md#how-to-build-and-run-samples) so the gap is purely in the README.md file which does not mention ninja.

The README files of the other 7 samples need to be updated to match helloworld and host_verify.

### PR DESCRIPTION
**Pull request by radhikaj**
This PR fixes #3522

1. The samples folder had two files- README_Linux.md and README_Windows.md. These got renamed to README.md during installation. This created a problem where the folders inside samples would refer to README.md which would work when the samples were installed on a VM, but the links would be invalid while reading the sample docs on our github repo. I have fixed this by having a single README.md and then splitting that off into BuildSamplesWindows.md and BuildSamplesLinux.md. The files will be installed and the names will stay the same both on the githubrepo and in the installation.
2. All the Linux build information is aggregated into BuildSamplesLinux.md(This file was created by cutting and pasting off of README_Linux.md) under samples/. Similarly, for Windows, this is aggregated into BuildSamplesWindows.md(created by cutting and pasting off of README_Windows.md) under samples/. The individual samples then redirect to these for building. This removes redundant informaiton.
3. The LVI documentation will continue to live under helloworld since this is the sample showcasing how to build and run with LVI mitigations.

### CODE REVIEW

**Review by mingweishih — APPROVED** [review]

File: samples/BuildSamplesLinux.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+~/openenclave/share/openenclave/samples/helloworld$ make run
+```
+
+### Build the samples using CMake
+
+To build a sample using CMake, change directory to your target sample directory and execute the following commands:
+
+mkdir build && cd build
+cmake ..
+make
```
[the remark this answers is not in the document]
Comment (radhikaj) [reply samples/BuildSamplesLinux.md:83]: Fixed, Thanks!

File: samples/helloworld/README.md [hunk 2]
```diff
@@ -66,6 +67,10 @@ To generate the marshaling code the untrusted host uses to call into the trusted
 oeedger8r ../helloworld.edl --untrusted
 ```
 
+```cmd
+oeedger8r ..\helloworld.edl --untrusted
+```
+
 This command compiles the `helloworld.edl` file and generates the following files within the host directory:
 
 | file | description |
```
[the remark this answers is not in the document]
Comment (radhikaj) [reply samples/helloworld/README.md:76]: I removed the Makefile section at 233.

[the remark this answers is not in the document]
Comment (radhikaj) [reply samples/helloworld/README.md:76]: I moved the makefile -build, clean, run to BuildSamplesLinux.md

[the remark this answers is not in the document]
Comment (radhikaj) [reply samples/helloworld/README.md:76]: I have modified the helloworld to add Windows commands for each Linux command.

File: samples/helloworld/README.md [hunk 3]
```diff
@@ -66,6 +67,10 @@ To generate the marshaling code the untrusted host uses to call into the trusted
 oeedger8r ../helloworld.edl --untrusted
 ```
 
+```cmd
+oeedger8r ..\helloworld.edl --untrusted
+```
```
[the remark this answers is not in the document]
Comment (radhikaj) [reply samples/helloworld/README.md:72]: I fixed this so I have Windows and Linux variats.

File: samples/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# Open Enclave SDK Samples
+
+All the samples that come with the Open Enclave SDK installation share a similar directory structure and build instructions. This document describes how to setup, build, sign, and run these samples.
+
+## Common Sample information
+
+### How Sample source code directories were structured
+
+Open Enclave SDK helps developers build enclave applications. An enclave application is partitioned into an untrusted component (called a host) and a trusted component (called an enclave). An enclave is a secure container whose memory (text and data) is protected from access by outside entities, including the host, privileged users, and even the hardware. All functionality that needs to be run in a Trusted Execution Environment (TEE) should be compiled into the enclave binary. The enclave may run in an untrusted environment with the expectation that secrets will not be compromised. A host is a normal user mode application that loads an enclave into its address space before starting interacting with an enclave.
```
[the remark this answers is not in the document]
Comment (radhikaj) [reply samples/README.md]: I split this into 3 paras.

**Comment by radhikaj** [comment]
@dthaler, addressed your comments. Please review at your convenience.

File: samples/attested_tls/README.md [hunk 5]
```diff
@@ -68,35 +68,23 @@ Note: Both of them can run on the same machine or separate machines.
...
-```
 ### Running attested TLS server in loop
 By default the server exits after completing a TLS session with a client. `-server-in-loop` run-time option changes this behavior to allow the TLS server to handle multiple client requests.
+
+On Linux:
+
 ```bash
 ./server/host/tls_server_host ./server/enc/tls_server_enc.signed -port:12341 -server-in-loop
 or
 make run-server-in-loop
```
[the remark this answers is not in the document]
Comment (radhikaj) [reply samples/attested_tls/README.md:83]: Is the suggestion to do it in this PR?

**Comment by radhikaj** [comment]
@dthaler , thanks for reviewing. I added CMakeLists.txt to the list of files that come with the samples.

**Review by dthaler — APPROVED** [review]
Ok modulo two trivial nits

**Comment by radhikaj** [comment]
Thanks for reviewing @dthaler and @mingweishih

**Comment by radhikaj** [comment]
Fixes #3501

**Comment by radhikaj** [comment]
Bors r+

**Comment by BRMcLaren** [comment]
bors try

File: samples/attested_tls/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (dthaler) [reply samples/attested_tls/README.md:83]: It's ok with me either way.  I was thinking we should file a separate issue.

**Pull request merged**